nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -217,22 +232,29 @@ class NodeletLazy: public nodelet::Nodelet
     */
   template<class T> ros::Publisher
   advertise(ros::NodeHandle& nh,
-            std::string topic, int queue_size, bool latch=false)
+            const std::string& topic, int queue_size, bool latch=false)
```

Comment:
The change to `const &` is in a separate commit, so if you don't like it I can remove it from this PR, but we shouldn't be passing a `std::string` by copy.